server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -814,43 +815,71 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
         ByteBuffer schemaBinary = schemaInfo.getSchemaData();
         boolean isValid = true;
         String invalidityCause = "";
+        String type = schemaInfo.getType();
         try {
             String schemaString;
+            String[] tokens;
             switch (schemaInfo.getSerializationFormat()) {
                 case Protobuf:
-                    String[] tokens = NameUtil.extractNameAndQualifier(schemaInfo.getType());
-                    String qualifier = tokens[1]; // this can be empty string if schema info had no qualifier
-                    String name = tokens[0];
+                    tokens = NameUtil.extractNameAndQualifier(schemaInfo.getType());
+                    String protoPackage; 
+                    String protoName;
                     DescriptorProtos.FileDescriptorSet fileDescriptorSet = DescriptorProtos.FileDescriptorSet.parseFrom(
                             schemaInfo.getSchemaData());
+                    if (Strings.isNullOrEmpty(tokens[0])) {
```

Comment:
nit: It may be easier to understand the essence of the logic here if we use variables like these instead: 
```
String nameInSchemaInfo = tokens[0];
String qualifierInSchemaInfo = tokens[1]; 
```